tests/protocol/test_extensions_goal_state.py

```diff
@@ -34,6 +34,10 @@ def test_property(name, value):
             agent_manifests[0].family = 'MOCK_FAMILY'
             test_property("agent_manifests", agent_manifests)
 
+            extensions = copy.copy(from_vm_settings.extensions)
```

Comment:
Any reason for not using deepcopy here? Maybe keep them similar for consistency?

Reply:
should actually be a deep copy, thanks for pointing it out. right now this works because this is the last test in this method.

i am already working on the next change to parse the extension settings so I'll leave this PR as it is and do the change as part of the code I am working on.